arithmetization/src/test/java/net/consensys/linea/zktracer/precompiles/LowGasStipendPrecompileCallTests.java

```diff
@@ -0,0 +1,327 @@
+        || (precompileAddress.equals(ALTBN128_ADD)
+            && value > 0) // precompileCost is 150 but stipend is at least 2300 so gas cannot be
+    // insufficient
+    ) {
+      assertFalse(insufficientGasForPrecompile);
+    } else {
+      assertTrue(insufficientGasForPrecompile);
+    }
+  }
+
+  static Stream<Arguments> lowGasStipendPrecompileCallTestSource() {
+    List<Arguments> arguments = new ArrayList<>();
+    for (GasCase gasCase : GasCase.values()) {
+      for (ArgumentCase argumentCase : ArgumentCase.values()) {
+        arguments.add(Arguments.of(ECREC, argumentCase, gasCase, false));
+        arguments.add(Arguments.of(SHA256, argumentCase, gasCase, false));
+        arguments.add(Arguments.of(RIPEMD160, argumentCase, gasCase, false));
+        arguments.add(Arguments.of(ID, argumentCase, gasCase, false));
+        arguments.add(Arguments.of(ALTBN128_ADD, argumentCase, gasCase, false));
+        arguments.add(Arguments.of(ALTBN128_MUL, argumentCase, gasCase, false));
+        arguments.add(Arguments.of(Address.ALTBN128_PAIRING, argumentCase, gasCase, false));
+        arguments.add(Arguments.of(BLAKE2B_F_COMPRESSION, argumentCase, gasCase, false));
+      }
+      // The NON_ZERO for MODEXP case will be treated in a separate test
+      arguments.add(Arguments.of(MODEXP, ArgumentCase.ZERO, gasCase, false));
+      arguments.add(Arguments.of(MODEXP, ArgumentCase.ZERO, gasCase, true));
+    }
+    return arguments.stream();
+  }
+
+  // Support methods
+  private static void prepareBlake2F(
+      BytecodeCompiler program, int rLeadingByte, int callDataOffset) {
+    program
+        .push(rLeadingByte) // For simplicity, we only set the first byte of r
+        .push(callDataOffset + 2) // offset
+        // Writing rLeadingByte at this offset
+        // allows to have r = 0x00000000 or r = 0x00001200
+        .op(OpCode.MSTORE8);
+  }
+
+  private void prepareSha256Ripemd160Id(BytecodeCompiler program, int nWords, int callDataOffset) {
+    populateMemory(program, nWords, callDataOffset);
+  }
+
+  private static BigInteger prepareModexp(
+      int bbs, int mbs, int ebs, int callDataSize, BytecodeCompiler program) {
+    final Bytes32 bbsPadded = Bytes32.leftPad(Bytes.of(bbs));
+    final Bytes32 ebsPadded = Bytes32.leftPad(Bytes.of(ebs));
+    final Bytes32 mbsPadded = Bytes32.leftPad(Bytes.of(mbs));
+    final Bytes32 bemPadded =
+        Bytes32.rightPad(
+            Bytes.fromHexString("0x" + "aa".repeat(bbs) + "ff".repeat(ebs) + "bb".repeat(mbs)));
+    // Note that is an arbitrary value respecting bbs, ebs, mbs
+
+    program
+        .push(bbsPadded)
+        .push(0) // offset
+        .op(OpCode.MSTORE)
+        .push(ebsPadded)
+        .push(32) // offset
+        .op(OpCode.MSTORE)
+        .push(mbsPadded)
+        .push(64) // offset
+        .op(OpCode.MSTORE)
+        .push(bemPadded)
+        .push(96) // offset
+        .op(OpCode.MSTORE);
```

Comment:
This isn't the right way to do it. The issue is that you are confined to testing short data because you can only `MSTORE` a 32B integer. So `bemPadded` has to be at most 32 bytes long. However it seems that your own scenarios violate this condition, e.g. when `modexpCostGT200 = true` you get
```
bbs + ebs + mbs = 0x26 = 38 > 32
```

Reply:
You should use the technique I sent you guys over slack and which you'll find in my precompile PR. This avoids having to prepare data and `MSTORE`'ing it. You just `EXTCODECOPY` the stuff right into memory from an account that's been prepared to contain exactly the string you want.